test/pkg/environment/common/expectations.go

```diff
@@ -91,6 +91,26 @@ func (env *Environment) ExpectUpdated(objects ...client.Object) {
 	}
 }
 
+// WARNING: This ignores the resource version check, which can result in
+// overwriting changes made by other controllers in the cluster.
+// This is useful in ensuring that we can clean up resources by patching
+// out finalizers.
+// Grab the object before making the updates to reduce the chance of this race.
+func (env *Environment) ExpectStatusUpdated(objects ...client.Object) {
```

Comment:
nit: We should eventually think about which of these we can migrate over into operatorpkg to reduce some of our code duplication here

Reply:
Absolutely, once the migration is over. We will be using the common environment package from here